commit b7f31602eefb908b048d59feb7e7078c0780a7cb
Author: Roman Rybalko <devel@romanr.info>
Date:   Thu Dec 25 18:34:55 2014 +0300

    $host_port variable (deliver_port)

diff --git src/deliver.c src/deliver.c
index 87b54d8..a4e506a 100644
--- src/deliver.c
+++ src/deliver.c
@@ -141,11 +141,13 @@ the first address. */
 if (addr->host_list == NULL)
   {
   deliver_host = deliver_host_address = US"";
+  deliver_port = 0;
   }
 else
   {
   deliver_host = addr->host_list->name;
   deliver_host_address = addr->host_list->address;
+  deliver_port = (addr->host_list->port == PORT_NONE) ? 25 : addr->host_list->port;
   }
 
 deliver_recipients = addr;
diff --git src/expand.c src/expand.c
index 70d7c7d..8516d1a 100644
--- src/expand.c
+++ src/expand.c
@@ -501,6 +501,7 @@ static var_entry var_table[] = {
   { "host_data",           vtype_stringptr,   &host_data },
   { "host_lookup_deferred",vtype_int,         &host_lookup_deferred },
   { "host_lookup_failed",  vtype_int,         &host_lookup_failed },
+  { "host_port",           vtype_int,         &deliver_port },
   { "inode",               vtype_ino,         &deliver_inode },
   { "interface_address",   vtype_stringptr,   &interface_address },
   { "interface_port",      vtype_int,         &interface_port },
diff --git src/globals.c src/globals.c
index d3f9987..15a8ede 100644
--- src/globals.c
+++ src/globals.c
@@ -588,6 +588,7 @@ uschar *deliver_localpart_suffix = NULL;
 BOOL    deliver_force_thaw     = FALSE;
 BOOL    deliver_manual_thaw    = FALSE;
 uschar *deliver_out_buffer     = NULL;
+int     deliver_port           = -1;
 int     deliver_queue_load_max = -1;
 address_item  *deliver_recipients = NULL;
 uschar *deliver_selectstring   = NULL;
diff --git src/globals.h src/globals.h
index 2bedcf5..c07bd40 100644
--- src/globals.h
+++ src/globals.h
@@ -339,6 +339,7 @@ extern uschar *deliver_localpart_suffix; /* The stripped suffix, if any */
 extern BOOL    deliver_force_thaw;     /* TRUE to force thaw in queue run */
 extern BOOL    deliver_manual_thaw;    /* TRUE if manually thawed */
 extern uschar *deliver_out_buffer;     /* Buffer for copying file */
+extern int     deliver_port;           /* Host-specific port for delivery */
 extern int     deliver_queue_load_max; /* Different value for queue running */
 extern address_item *deliver_recipients; /* Current set of addresses */
 extern uschar *deliver_selectstring;   /* For selecting by recipient */
diff --git src/transports/smtp.c src/transports/smtp.c
index 40eebe8..9bf51e6 100644
--- src/transports/smtp.c
+++ src/transports/smtp.c
@@ -3035,6 +3035,7 @@ for (cutoff_retry = 0; expired &&
 
     deliver_host = host->name;
     deliver_host_address = host->address;
+    deliver_port = (host->port == PORT_NONE) ? port : host->port;
     lookup_dnssec_authenticated = host->dnssec == DS_YES ? US"yes"
 				: host->dnssec == DS_NO ? US"no"
 				: US"";
