diff --git src/tls-openssl.c src/tls-openssl.c
index 18994ea..53cec0a 100644
--- src/tls-openssl.c
+++ src/tls-openssl.c
@@ -279,6 +279,10 @@ static uschar txt[256];
 
 X509_NAME_oneline(X509_get_subject_name(cert), CS txt, sizeof(txt));
 
+if (tlsp->peercert)
+  X509_free(tlsp->peercert);
+tlsp->peercert = X509_dup(cert);
+
 if (state == 0)
   {
   log_write(0, LOG_MAIN, "SSL verify error: depth=%d error=%s cert=%s",
@@ -289,7 +293,6 @@ if (state == 0)
   *calledp = TRUE;
   if (!*optionalp)
     {
-    tlsp->peercert = X509_dup(cert);
     return 0;			    /* reject */
     }
   DEBUG(D_tls) debug_printf("SSL verify failure overridden (host in "
@@ -319,7 +322,6 @@ else
 #endif
 
   tlsp->peerdn = txt;
-  tlsp->peercert = X509_dup(cert);
 
 #ifdef EXPERIMENTAL_CERTNAMES
   if (  tlsp == &tls_out
